pkg/util/time.go

```diff
@@ -98,3 +99,34 @@ func NewDisableableTicker(interval time.Duration) (func(), <-chan time.Time) {
 	tick := time.NewTicker(interval)
 	return func() { tick.Stop() }, tick.C
 }
+
+// JSONSecondsTimestamp is time.Time, but stored as Unix seconds integer value when marshalled to JSON.
+// Zero time.Time value is marshalled as 0.
+type JSONSecondsTimestamp time.Time
+
+func (t JSONSecondsTimestamp) MarshalJSON() ([]byte, error) {
+	if t.Time().IsZero() {
+		return []byte("0"), nil
+	}
+	return []byte(strconv.FormatInt(time.Time(t).Unix(), 10)), nil
+}
+
+func (t *JSONSecondsTimestamp) UnmarshalJSON(data []byte) error {
+	i, err := strconv.ParseInt(string(data[:]), 10, 64)
+	if err != nil {
+		return err
+	}
+	switch {
+	case i == 0:
+		*t = JSONSecondsTimestamp(time.Time{})
```

Comment:
isn't `i==0` equivalent to `time.Unix(0, 0)`?

Reply:
Unfortunately not.

* `time.Unix(0, 0)` is `1970-01-01 00:00:00 +0000 UTC`.
* `time.Time{}` is `0001-01-01 00:00:00 +0000 UTC`

Reply:
so if the json doc contains `"time": 0` we treat this as `0001-01-01`, but it could have been intended as `1970-01-01` (which is also valid)

Reply:
and also now the code changes what it receives

```
0001-01-01 --unmarshal--> 0 --marshal--> 1970-01-01
```

it started with one date, but ended up with another

Reply:
Yes, you're correct and your examples are valid. Given the special nature of both values, I don't think this is a showstopper though.